documentation/metrics-api/v2/metrics_api.markdown

```diff
@@ -0,0 +1,182 @@
+---
+layout: default
+title: "Puppet Server: Metrics API v2"
+canonical: "/puppetserver/latest/metrics-api/v2/metrics_api.html"
+---
+
+By default, Puppet Server enables two optional web APIs for
+[Java Management Extension (JMX)](https://docs.oracle.com/javase/tutorial/jmx/index.html)
+metrics, namely
+[managed beans (MBeans)](https://docs.oracle.com/javase/tutorial/jmx/mbeans/). For the older metrics API, see [the `/metrics/v1` documentation](../v1/metrics_api.html).
+
+## Jolokia endpoints
+
+The v2 metrics endpoint uses the [Jolokia](https://jolokia.org) library, an extensive open-source metrics library with its own documentation.
+
+The documentation below provides only the information you need to use the metrics as configured by default for Puppet Server, but Jolokia offers more features than are described below. Consult the [Jolokia documentation](https://jolokia.org/documentation.html) for more information.
+
+For security reasons, we enable only the read-access Jolokia interface by default:
+
+-   `read`
+-   `list`
+-   `version`
+-   `search`
+
+### Configuring Jolokia
+
+To change the security access policy, create the `/etc/puppetlabs/puppetserver/jolokia-access.xml` file with contents that follow the [Jolokia access policy](https://jolokia.org/reference/html/security.html) and uncomment the `metrics.metrics-webservice.jolokia.servlet-init-params.policyLocation` parameter before restarting puppetserver.
+
+The `metrics.metrics-webservice.jolokia.servlet-init-params` table
+within the [`/etc/puppetlabs/puppetserver/conf.d/metrics.conf`](../../config_file_metrics.markdown) file provides more configuration options. See Jolokia's [agent initialization documentation](https://jolokia.org/reference/html/agents.html#agent-war-init-params) for all of the available options.
```

Comment:
oy vey, it looks like this file doesn't exist? And that the metrics.conf file isn't documented anywhere (PE [links to how to configure graphite via hiera in the console](https://docs.puppet.com/pe/latest/config_puppetserver.html#puppet-server-configuration-files) and tk-metrics isn't [much more than the default config file](https://github.com/puppetlabs/trapperkeeper-metrics/blob/master/documentation/configuration.md))...

I assume I should probably pull in the tk-metrics configuration file and annotate it some with the equivalent file based recommendations for configuring the graphite reporter from the PE docs. Would that be sufficient? Sorry, about that.